tests/scripts/thread-cert/node.py

```diff
@@ -2727,27 +2723,52 @@ def discover_mdns_service(self, instance, name, host_name, timeout=2):
         The return value is a dict with the same key/values of srp_server_get_service
         except that we don't have a `deleted` field here.
         """
-        zeroconf = Zeroconf()
-        timeout *= 1000  # Zeroconf use timeout in milliseconds.
-        try:
-            info = ServiceInfo(type_=f'{name}.local.', name=f'{instance}.{name}.local.', server=f'{host_name}.local.')
-            while timeout > 0 and not info.parsed_addresses():
-                info.request(zeroconf, 500)
-                timeout -= 500
-            if info.parsed_addresses():
-                return {
-                    'fullname': info.name,
-                    'instance': info.get_name(),
-                    'name': name,
-                    'port': info.port,
-                    'weight': info.weight,
-                    'priority': info.priority,
-                    'host_fullname': info.server,
-                    'host': host_name,
-                    'addresses': info.parsed_addresses()
-                }
-        finally:
-            zeroconf.close()
+
+        self.bash(f'dns-sd -Z {name} local. > /tmp/{name} 2>&1 &')
+        self.bash(f'dns-sd -G v6 {host_name}.local. > /tmp/{host_name} 2>&1 &')
+        time.sleep(timeout)
+
+        self.bash(f'pkill dns-sd')
+        addresses = []
+        service = {}
+
+        print(self.bash(f'cat /tmp/{host_name}'))
+        print(self.bash(f'cat /tmp/{name}'))
+
+        # example output in the host file:
+        # Timestamp     A/R Flags if Hostname                               Address                                     TTL
+        # 9:38:09.274  Add     23 48 my-host.local.                         2001:0000:0000:0000:0000:0000:0000:0002%<0>  120
+        #
+        for line in self.bash(f'cat /tmp/{host_name}'):
+            elements = line.split()
+            if not elements or len(elements) < 6 or not elements[4].startswith(host_name):
+                continue
+            addresses.append(elements[5].split('%')[0])
+
+        print(addresses)
```

Comment:
```suggestion
```
remove debugging prints ?

Reply:
It will be helpful if the test case fails.